File: src/main/java/com/dashboard/repositories/TaskRepository.java
package com.dashboard.repositories;

import com.dashboard.models.JPA.Task;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;

public interface TaskRepository extends JpaRepository<Task, Integer> {
    List<Task> findByPriority(String priority);
    List<Task> findTasksByStatus(String status);
}

File: src/main/kotlin/dashboard/repositories/TaskRepository.kt
package dashboard.repositories

import dashboard.models.jpa.Task
import org.springframework.data.jpa.repository.JpaRepository

interface TaskRepository : JpaRepository<Task, Int> {
    fun findByPriority(priority: String): List<Task>
    fun findTasksByStatus(status: String): List<Task>
}